src/client/cli/formatter/yaml_formatter.cpp

```diff
@@ -58,92 +58,140 @@ std::map<std::string, YAML::Node> format_images(const ImageInfo& images_info)
 
-        YAML::Node disk;
-        disk["used"] = instance_details.disk_usage().empty() ? YAML::Node() : YAML::Node(instance_details.disk_usage());
-        disk["total"] = info.disk_total().empty() ? YAML::Node() : YAML::Node(info.disk_total());
+            mount_node["gid_mappings"].push_back(
+                fmt::format("{}:{}", std::to_string(host_gid),
+                            (instance_gid == mp::default_id) ? "default" : std::to_string(instance_gid)));
+        }
 
-        // TODO: disk name should come from daemon
-        YAML::Node disk_node;
-        disk_node["sda1"] = disk;
-        instance_node["disks"].push_back(disk_node);
+        mount_node["source_path"] = mount.source_path();
+        mounts[mount.target_path()] = mount_node;
+    }
+    instance_node["mounts"] = mounts;
 
-        YAML::Node memory;
-        memory["usage"] = instance_details.memory_usage().empty()
-                              ? YAML::Node()
-                              : YAML::Node(std::stoll(instance_details.memory_usage()));
-        memory["total"] = info.memory_total().empty() ? YAML::Node() : YAML::Node(std::stoll(info.memory_total()));
-        instance_node["memory"] = memory;
+    return instance_node;
+}
 
-        instance_node["ipv4"] = YAML::Node(YAML::NodeType::Sequence);
-        for (const auto& ip : instance_details.ipv4())
-            instance_node["ipv4"].push_back(ip);
+YAML::Node generate_instance_info_report(const mp::InfoReply& reply)
+{
+    YAML::Node info_node;
+
+    info_node["errors"].push_back(YAML::Null);
 
-        YAML::Node mounts;
-        for (const auto& mount : info.mount_info().mount_paths())
+    for (const auto& info : mp::format::sort_instances_and_snapshots(reply.detailed_report().details()))
+    {
+        if (info.has_instance_info())
         {
-            YAML::Node mount_node;
-
-            for (const auto& uid_mapping : mount.mount_maps().uid_mappings())
-            {
-                auto host_uid = uid_mapping.host_id();
-                auto instance_uid = uid_mapping.instance_id();
-
-                mount_node["uid_mappings"].push_back(
-                    fmt::format("{}:{}", std::to_string(host_uid),
-                                (instance_uid == mp::default_id) ? "default" : std::to_string(instance_uid)));
-            }
-            for (const auto& gid_mapping : mount.mount_maps().gid_mappings())
-            {
-                auto host_gid = gid_mapping.host_id();
-                auto instance_gid = gid_mapping.instance_id();
-
-                mount_node["gid_mappings"].push_back(
-                    fmt::format("{}:{}", std::to_string(host_gid),
-                                (instance_gid == mp::default_id) ? "default" : std::to_string(instance_gid)));
-            }
-
-            mount_node["source_path"] = mount.source_path();
-            mounts[mount.target_path()] = mount_node;
+            info_node[info.name()].push_back(generate_instance_details(info));
```

Comment:
Ugh, why is this an array...? Anyway, I realize you're not the one introducing it and it's there since forever.